### PR TITLE
Enable Django for VS

### PR DESCRIPTION
**Pull request by karthiknadig**
In this PR:
1. Parse launch options sent by VS
2. Enable Django and Jinja2 template debugging for VS

Notes:
For VS flask project we have to enable breakpoints in flask project, and send `FLASK_DEBUG=True` as debug option.

Pending:
- [x] Add tests to include launch options as sent by VS

### CODE REVIEW

File: ptvsd/wrapper.py [hunk 1]
```diff
@@ -602,6 +602,7 @@ def __init__(self, socket, pydevd, logfile=None, killonclose=True):
         self.modules_mgr = ModulesManager(self)
         self.disconnect_request = None
         self.launch_arguments = None
+        self.launch_options = {}
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: I'd check this to debug options. For instance when attaching to Django application we'll need to option DJANGO_DEBUG=True in the attach request for brains to work.

File: ptvsd/wrapper.py [hunk 2]
```diff
@@ -790,6 +791,30 @@ def process_launch_arguments(self):
...
+    def __parse_launch_option_value(self, value):
+        if value == 'True' or value == 'False':
+            return bool(value)
+        return unquote(value)
+
+    def __parse_launch_options(self, launch_options):
+        """VS options semicolon separated key=value pairs
+            WAIT_ON_ABNORMAL_EXIT=True|False
+            WAIT_ON_NORMAL_EXIT=True|False
+            REDIRECT_OUTPUT=True|False
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: You're not handling REDIRECT_OUTPUT in launch request.

File: ptvsd/wrapper.py [hunk 3]
```diff
@@ -1252,30 +1279,36 @@ def on_setBreakpoints(self, request, args):
         self.path_casing.track_file_path_case(path)
         src_bps = args.get('breakpoints', [])
 
+        bp_type = 'python-line'
+        if not path.lower().endswith('.py'):
+            if self.launch_options.get('DJANGO_DEBUG', False):
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: This doesn't apply to launch debugging only, attaching too would need this check, hence the suggestion to change property name

File: tests/highlevel/test_messages.py [hunk 4]
```diff
@@ -1059,6 +1057,80 @@ def test_multiple_files(self):
                 '2\tpython-line\teggs.py\t17\tNone\tNone\tNone'),
         ])
 
+    def test_vs_django(self):
+        with self.launched(args={'options': 'DJANGO_DEBUG=True'}):
+            self.send_request(
```
Comment (DonJayamanne) [inline tests/highlevel/test_messages.py]: We will also need tests for remote debugging (attach request).

File: ptvsd/wrapper.py [hunk 5]
```diff
@@ -790,6 +791,30 @@ def process_launch_arguments(self):
             redirect_output = ''
         self.pydevd_request(pydevd_comm.CMD_REDIRECT_OUTPUT, redirect_output)
 
+    def __parse_launch_option_value(self, value):
+        if value == 'True' or value == 'False':
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: We need to add support for attaching as well. Or we could make that a separate PR.

File: ptvsd/wrapper.py [hunk 6]
```diff
@@ -790,6 +791,30 @@ def process_launch_arguments(self):
...
+            return bool(value)
+        return unquote(value)
+
+    def __parse_launch_options(self, launch_options):
+        """VS options semicolon separated key=value pairs
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: I'd remove this message. There's no reason VSC or another debugger can't use this format. If it did then one would assume this was being used only by VS.

**Comment by DonJayamanne** [comment]
Lets keep this PR simple, ignore remote debugging (attach), Please create an issue for that.

File: ptvsd/wrapper.py (hunk 2 above)
Comment (karthiknadig) [reply ptvsd/wrapper.py]: I want to keep this about DJANGO. created a bug to handle redirect separately: https://github.com/Microsoft/ptvsd/issues/250

File: ptvsd/wrapper.py (hunk 5 above)
Comment (karthiknadig) [reply ptvsd/wrapper.py]: I will handle attach separately: https://github.com/Microsoft/ptvsd/issues/248

File: tests/highlevel/test_messages.py (hunk 4 above)
Comment (karthiknadig) [reply tests/highlevel/test_messages.py]: Will do this as a part of attach: https://github.com/Microsoft/ptvsd/issues/248

File: ptvsd/wrapper.py (hunk 2 above)
Comment (DonJayamanne) [reply ptvsd/wrapper.py]: Hmm, bug or missing feature 😄
